### PR TITLE
[TwoAddressInstruction] Use isPlainlyKilled in processTiedPairs

### PR DESCRIPTION
**Pull request by jayfoad**
Calling isPlainlyKilled instead of directly checking for a kill flag
should make processTiedPairs behave the same with LiveIntervals
(i.e. when compiling with -early-live-intervals) as it does with
LiveVariables.

### CODE REVIEW

File: llvm/test/CodeGen/X86/combine-or.ll [hunk 1]
```diff
@@ -235,15 +236,26 @@ define <4 x i32> @test17(<4 x i32> %a, <4 x i32> %b) {
...
+;
+; CHECK-LIS-LABEL: test18:
+; CHECK-LIS:       # %bb.0:
+; CHECK-LIS-NEXT:    pxor %xmm2, %xmm2
+; CHECK-LIS-NEXT:    pxor %xmm3, %xmm3
+; CHECK-LIS-NEXT:    pblendw {{.*#+}} xmm3 = xmm0[0,1],xmm3[2,3,4,5,6,7]
+; CHECK-LIS-NEXT:    pshufd {{.*#+}} xmm0 = xmm3[1,0,1,1]
+; CHECK-LIS-NEXT:    pblendw {{.*#+}} xmm2 = xmm1[0,1],xmm2[2,3,4,5,6,7]
+; CHECK-LIS-NEXT:    por %xmm0, %xmm2
+; CHECK-LIS-NEXT:    movdqa %xmm2, %xmm0
```
Comment (jayfoad) [inline llvm/test/CodeGen/X86/combine-or.ll:257]: Regression here but I don't think TwoAddressInstruction is doing anything wrong. I can see RegisterCoalescer rematerializing a V_SET0 in a slightly unfortunate place which causes overlapping lifetimes for (a) one value that wants to use $xmm0 on input and (b) another value that wants to use $xmm0 on return, so one of them has to pick a different register and that causes the extra COPY.

File: llvm/test/CodeGen/SystemZ/rot-02.ll [hunk 2]
```diff
@@ -2,7 +2,8 @@
 ; Test removal of AND operations that don't affect last 6 bits of rotate amount
 ; operand.
 ;
-; RUN: llc < %s -mtriple=s390x-linux-gnu | FileCheck %s
+; RUN: llc < %s -mtriple=s390x-linux-gnu | FileCheck %s -check-prefixes=CHECK,CHECK-LV
+; RUN: llc < %s -mtriple=s390x-linux-gnu -early-live-intervals | FileCheck %s -check-prefixes=CHECK,CHECK-LIS
```
Comment (RKSimon) [inline llvm/test/CodeGen/SystemZ/rot-02.ll:6]: These changes look redundant?

File: llvm/test/CodeGen/X86/combine-rotates.ll [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+sse2 | FileCheck %s --check-prefixes=CHECK,SSE2
+; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+sse2 | FileCheck %s --check-prefixes=CHECK,SSE2,SSE2-LV
+; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+sse2 -early-live-intervals | FileCheck %s --check-prefixes=CHECK,SSE2,SSE2-LIS
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/combine-rotates.ll:3]: redundant change?

File: llvm/test/CodeGen/X86/combine-or.ll (hunk 1 above)
Comment (RKSimon) [reply llvm/test/CodeGen/X86/combine-or.ll:257]: We have similar issues with multiple zero vector registers, so normally I'd say this is acceptable but I'm worried this change might be encouraging the underlying problem?

**Review by RKSimon — COMMENTED** [review]
Please can you update the patch title/description?

File: llvm/test/CodeGen/SystemZ/rot-02.ll (hunk 2 above)
Comment (jayfoad) [reply llvm/test/CodeGen/SystemZ/rot-02.ll:6]: Ugh. I am still struggling with how to deal with stacked patches. In the first commit the separate prefixes are necessary. In the second patch, codegen improves and they are no longer needed.

File: llvm/test/CodeGen/X86/combine-or.ll (hunk 1 above)
Comment (jayfoad) [reply llvm/test/CodeGen/X86/combine-or.ll:257]: I saw equal and opposite changes of this sort in `combine-or.ll` and `combine-rotates.ll` so I suspect it is neither better nor worse, but I'm not sure how to prove that.

**Comment by jayfoad** [comment]
The first commit (adding test coverage) is now #66058.

For this PR, please only review the second commit.

**Comment by jayfoad** [comment]
Rebased. Ping!

File: llvm/test/CodeGen/X86/combine-rotates.ll [hunk 4]
```diff
@@ -477,3 +451,6 @@ declare i5 @llvm.fshl.i5(i5, i5, i5)
 
 declare <4 x i32> @llvm.fshl.v4i32(<4 x i32>, <4 x i32>, <4 x i32>)
 declare <4 x i32> @llvm.fshr.v4i32(<4 x i32>, <4 x i32>, <4 x i32>)
+;; NOTE: These prefixes are unused and the list is autogenerated. Do not add tests below this line:
+; SSE2-LIS: {{.*}}
+; SSE2-LV: {{.*}}
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/combine-rotates.ll:456]: Remove these prefixes from the RUN lines?

Comment (jayfoad) [reply llvm/test/CodeGen/X86/combine-rotates.ll]: Done.

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**